uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -300,7 +343,10 @@ define([
             return loadStatuses();
           })
           .then(function () {
-            return $q.all[initUserRole(), initOpenMode()];
+            return $q.all[
```

Comment:
the `initController` has become too big, almost 60 lines of code with 6 `then`. I think now its btter to break it into multiple smaller chunks of private functions.
```
 initAbsenceType();
            initStatus();
            initContact();

            if (vm.isMode('edit')) {
              initialLeaveRequestAttributes = vm.leaveRequest.attributes();
            }
```

```
initAbsencePeriod();
            setMinMax();

            return $q.all([
              loadAbsenceTypes(),
              loadCalendar()
            ]);
```
can be moved to private functions. Or any other suggestions? @dhavalsharma @AkA84